### ISSUE TITLE
configparser readfp() deprecated in 3.2

### ISSUE THREAD

**Issue by jmacego** [jdowner/gist#58]
/Users/macdj033/Library/Python/3.7/bin/gist:295: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
  config.readfp(fp)

Likely should get bundled with: https://github.com/jdowner/gist/issues/24

**Comment by jdowner** [comment]
Thanks for bringing this up John, and I agree that it should be bundled with #24. Something that is long overdue.

**Comment by jdowner** [comment]
@jmacego I am running python3.6.5 and I do not see the deprecation warning. I have pushed a fix to the branch 'config'. Would you mind trying it out to see if that addressing the issue for you?

**Comment by jmacego** [comment]
@jdowner at least for 3.7.2 your config branch looks ready to merge and distribute and so forth.

I have another bug to log, but probably won't get to it until Monday if you want any code/pull request along with it.

**Comment by engineertree5** [comment]
Hello gist,

not sure if my issue the exact same as jmacego but I could not find anything related to what I'm experiencing on the web. The below is what I'm experiencing when I run `git list`.

```
/usr/local/bin/gist:295: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
  config.readfp(fp)
1553126730.763 ERROR[gist] string indices must be integers
```

I double checked that I had the proper dependencies as well and still getting this error. The `DeprecationWarning` I can live with but its the `1553126730.763 ERROR[gist] string indices must be integers` that is causing the problem of letting me use your gist module. Was this working before, yes. Did anything change on my system, no. I'm not aware of what is needed for me to troubleshoot this but any information would be helpful. I'm working to become a better engineer and solving problems just like this one is half the battle

**Comment by jdowner** [comment]
Hi @engineertree5,

can you tell me what platform you are running on and what version of gist you are using? Also, to narrow down why this error could be happening, can you run,

```
curl  https://api.github.com/gists?access_token=<your-personal-token>
```

and make sure that you are getting a structure containing your gists.

**Comment by engineertree5** [comment]
Hey @jdowner,


.... So this is embarrassing. The issue was on my end (no surprise here). I tried the curl command you suggested.

```curl  https://api.github.com/gists?access_token=<your-personal-token>```

after executing this I kept getting:

```
{
  "message": "Bad credentials",
  "documentation_url": "https://developer.github.com/v3"
}
```

I went through my configuration and everything *seemed* to check out. Then I remembered I updated some .gitconfig credentials when I was doing some testing with git and some other editors. Long story short it ended up being my credentials. 

I will say it is good to know how to troubleshoot issues like this. I now know python-gist uses the https://api.github.com/v3 and is authenticating using our PAT within the .gist file. I'm going to pick apart this application and see what else I can learn and understand. Thanks for the response. You have no idea how helpful that comment was to me
